src/esp/gfx/DrawableGroup.cpp

```diff
@@ -0,0 +1,66 @@
+// Copyright (c) Facebook, Inc. and its affiliates.
+// This source code is licensed under the MIT license found in the
+// LICENSE file in the root directory of this source tree.
+#include "DrawableGroup.h"
+#include "Drawable.h"
+
+namespace esp {
+namespace gfx {
+
+class Drawable;
+DrawableGroup& DrawableGroup::add(Drawable& drawable) {
+  registerDrawable(drawable);
+  this->Magnum::SceneGraph::DrawableGroup3D::add(drawable);
+  return *this;
+}
+
+DrawableGroup& DrawableGroup::remove(Drawable& drawable) {
+  unregisterDrawable(drawable);
+  this->Magnum::SceneGraph::DrawableGroup3D::remove(drawable);
+  return *this;
+}
+
+DrawableGroup::~DrawableGroup() {
+  // has to unregister any currently existing drawables
+  for (int iDrawable = 0; iDrawable < this->size(); ++iDrawable) {
+    unregisterDrawable(static_cast<Drawable&>((*this)[iDrawable]));
+  }
+}
+
+bool DrawableGroup::hasDrawable(uint64_t id) const {
+  return (idToDrawable_.find(id) != idToDrawable_.end());
+}
+
+Drawable* DrawableGroup::getDrawable(uint64_t id) const {
+  if (hasDrawable(id)) {
+    return idToDrawable_.at(id);
+  }
```

Comment:
Unnecessary two map searches, could be one. I'm lazy to write another snippet -- use `find()` ;)

Reply:
This is an unordered_map not a map. So the search depends on the hash of the key, which is super fast. 

But I will change anyway.

Reply:
In theory it is fast, in practice `std::unordered_map` is [one of the worst performing implementations of a hashmap](https://tessil.github.io/2016/08/29/benchmark-hopscotch-map.html), due to various unfortunate design decisions mandated by the standard .. and in some cases the constant overhead of the O(1) lookup can be larger than a O(log n) search in a map.

doesn't matter if map or unordered_map, the change will make it 2x faster ;)